src/Orleans.Core/Runtime/AsynchAgent.cs

```diff
@@ -122,6 +122,7 @@ public virtual void Stop()
                     {
                         State = ThreadState.StopRequested;
                         Cts.Cancel();
+                        executor = null;
```

Comment:
This change is made in order to use fresh executor after Stop - Start iteration.